front/src/modules/trainschedule/components/ManageTrainSchedule/helpers/adjustConfWithTrainToModify.ts

```diff
@@ -33,6 +33,26 @@ function convertStepToPointOnMap(step?: ArrayElement<Path['steps']>): PointOnMap
     : undefined;
 }
 
+export function loadPathFinding(path: Path, dispatch: Dispatch) {
+  if (path.steps && path.steps.length > 1) {
+    dispatch(updatePathfindingID(path.id));
+    dispatch(updateItinerary(path));
+    dispatch(updateOrigin(convertStepToPointOnMap(path.steps[0])));
+    dispatch(updateDestination(convertStepToPointOnMap(path.steps.at(-1))));
+    if (path.steps.length > 2) {
+      const vias = path.steps
+        .filter(
+          (via, idx) => idx !== 0 && path.steps && idx < path.steps.length - 1 && !via.suggestion
+        )
+        .map((via) => convertStepToPointOnMap(via));
+      dispatch(replaceVias(compact(vias)));
+      dispatch(updateSuggeredVias(compact(path.steps.map((via) => convertStepToPointOnMap(via)))));
+    } else {
+      dispatch(replaceVias([]));
+    }
+  }
+}
```

Comment:
We could maybe either reuse some of the code use in `Pathfinding.tsx` for this function or use the code of this function in `Pathfinding.tsx` (even though some parts are specific to each components)

Reply:
I tried, a lot, but finally the modifications on Pathfinding.tsx were risky to add some new bug, and needed lot of rework & rewriting on other components based on... so I dropped the case, unfortunately.

When Pathfinding.tsx will be rewritten (and simplified), we'll have an exportable function to do this job, that'll permit to update pathfinding state on the interface. That's not the case today I'm afraid.

Reply:
Ok understood :)